addons/knowledge/controllers/main.py

```diff
@@ -79,32 +71,26 @@ def _prepare_articles_tree_values(self, active_article_id=False, unfolded_articl
         :param unfolded_articles: (list of ids) Used to display the children of the given article ids.
         :return: (Dict) that will be used to render templates in the articles tree side panel.
         """
-        # sudo article to avoid access error on member or on article for external users.
-        # The article the user can see will be based on user_has_access.
-        root_articles = request.env["knowledge.article"].sudo().search([
-            ("parent_id", "=", False), ('user_has_access', '=', True)]
-        ).sorted('sequence')
+        root_articles = request.env["knowledge.article"].search([("parent_id", "=", False)]).sorted('sequence')
 
         return {
             "active_article_id": active_article_id,
             "public_articles": root_articles.filtered(lambda article: article.category == 'workspace'),
             "shared_articles": root_articles.filtered(lambda article: article.category == 'shared'),
             "unfolded_articles": unfolded_articles,
-            "favourites": request.env['knowledge.article.favourite'].sudo().search([
-                ("user_id", "=", request.env.user.id), ('article_id.user_has_access', '=', True)
-            ]),
+            "favourites": request.env['knowledge.article.favourite'].search([("user_id", "=", request.env.user.id)]),
             # private_articles will be empty for share users
             "private_articles": root_articles.filtered(lambda article: article.owner_id == request.env.user)
         }
 
     @http.route('/knowledge/get_favourite_tree', type='json', auth='user')
     def get_favourite_tree(self, active_article_id=False):
-        favourite_articles = request.env['knowledge.article.favourite'].search([("user_id", "=", request.env.user.id), ('article_id.user_has_access', '=', True)])
+        favourite_articles = request.env['knowledge.article.favourite'].search([("user_id", "=", request.env.user.id)])
         return request.env['ir.qweb']._render('knowledge.article_favourite_section_template', {'favourites': favourite_articles, "active_article_id": active_article_id})
 
     @http.route('/knowledge/get_favourite_tree_frontend', type='json', auth='user')
     def get_favourite_tree_frontend(self, active_article_id=False):
-        favourite_articles = request.env['knowledge.article.favourite'].sudo().search([("user_id", "=", request.env.user.id), ('article_id.user_has_access', '=', True)])
+        favourite_articles = request.env['knowledge.article.favourite'].search([("user_id", "=", request.env.user.id)])
```

Comment:
je penses que l'à t'es qsuand meme obligé de le faire: parce que si tu n'as plus accès à un article que tu as favorite dans le passé, il sera tjs retournée ici, meme si t'as plus accès à l'article

Reply:
I don't think that's necessary, since in the access rules we have:
```xml
<record id="knowledge_user_article_favourites" model="ir.rule">
    <field name="name">Favourite articles</field>
    <field name="model_id" ref="model_knowledge_article_favourite"/>
    <field name="domain_force">[('user_id', '=', user.id), ('article_id.user_has_access', '=', True)]</field>
    <field name="groups" eval="[(4, ref('base.group_portal')), (4, ref('base.group_user'))]"/>
</record>
```
So the ones he doesn't have access to are filtered out